### PR TITLE
add argument factor to findCols

### PR DESCRIPTION
**Pull request by JosiahParry**
This PR adds an argument `factor` to `findCols()` which, when TRUE, returns the class indices as a factor object. From my explorations into the classInt package there is no "easy" way to return the interval labels along with the class indices. Having the labels would be particularly useful for visualization . Returning a factor enables users to have both the labels and the underlying integers. 

See the below reprex for an example use case of this PR.

``` r
data(jenks71, package="spData")

library(classInt)

x <- jenks71$jenks71
clI <- classIntervals(x, 5, "fisher")
jenks71$classes <- findCols(clI, factor = TRUE)

library(ggplot2)

ggplot(jenks71, aes(classes)) +
  geom_bar()
```

<sup>Created on 2022-06-06 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

### CODE REVIEW

**Comment by rsbivand** [comment]
Hi @JosiahParry : thanks! Are any more contributions forthcoming for **classInt**, or may I run reverse dependency checks?

**Comment by JosiahParry** [comment]
@rsbivand, thanks for merging! 

There is only one other thing which I've considered making a PR for, but would probably require your feedback. 

I was thinking of a simple function around `classInterval()` and `findCols()` so that labels can be created in one fell swoop. 

Perhaps a function such as `classify_intervals(var, method = "quantile", n = 5, factor = FALSE)` which would return a vector of of equal length as var. If you approve, I could make that PR quickly. Otherwise, then no, I don't have any further changes to make at the moment. 

Thanks again!

**Comment by rsbivand** [comment]
Yes, a wrapping function makes good sense, which could benefit downstream packages. the `classInterval()` function had didactic intentions when written, which was why the steps were made explicit.

File: R/classInt.R [hunk 1]
```diff
@@ -420,6 +420,11 @@ findCols <- function(clI)  {
   else {
 	cols <- apply(array(apply(outer(clI$var, clI$brks, ">"), 1, sum)), 1, max, 1)
   }
+  if (factor) {
+    col_vals <- names(tableClassIntervals(cols, clI$brks))
+    col_vals <- names(tableClassIntervals(cols, clI$brks))
```
Comment (kadyb) [inline R/classInt.R:425]: The above line is duplicated?

Comment (JosiahParry) [reply R/classInt.R:425]: Is it? 😜 Thanks for catching. Can you make an issue? Im out and about but can address later.

Comment (kadyb) [reply R/classInt.R:425]: No problem, I created [PR](https://github.com/r-spatial/classInt/pull/34) immediately.

**Comment by JosiahParry** [comment]
You rock, thanks!

**Pull request merged**